api/payments.go

```diff
@@ -276,18 +284,17 @@ func (a *API) PaymentRefund(ctx context.Context, w http.ResponseWriter, r *http.
 		return
 	}
 
-	provider := gcontext.GetPaymentProvider(ctx)
+	provider := gcontext.GetPaymentProviders(ctx)[order.PaymentProcessor]
```

Comment:
If that is empty, we are in trouble 😄. I can add a check.

What do you mean by "normalizing the param"?